core/include/vecmem/memory/resources/base_resource.hpp

```diff
@@ -0,0 +1,35 @@
+/** VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+#include "vecmem/memory/resources/memory_resource.hpp"
+
+namespace vecmem::resource {
+    /**
+     * @brief Virtual wrapper around std::pmr::memory_resource which exposes the
+     * host accessibility of the memory managed.
+     *
+     * The default memory_resource class included in the C++ standard is almost
+     * good enough for what we need, but we might want to have additional
+     * information about whether the memory managed is accessible by the host.
+     * This abstract class provides such functionality, and memory resource
+     * instances should ideally subclass this instead of memory_resource
+     * directly.
+     *
+     * @todo It may be difficult to determine the host accessibility for
+     * non-terminal memory resources, and we need a good solution for this.
+     */
+    class base_resource : public memory_resource {
+    public:
+        /**
+         * @brief Check whether this memory resource can be accessed by the host
+         * device.
+         */
+        virtual bool is_host_accessible() const = 0;
```

Comment:
I guess we're keeping this function after all... :confused:

I just still don't see where we'll be able to make use of it... Since `std::pmr::polymorphic_allocator` doesn't care about it. So even if we pass a "wrong type" of memory resource to a `vecmem::vector`, that will still not result in any clean failure.

So I would still just get rid of this class, and implement our resources directly on top of `vecmem::resource::memory_resource`. Which at that point I would really rather just call `vecmem::memory_resource`...

Reply:
I thought we discussed in our last meeting that we should keep this, but I guess I was wrong. I'll get rid of it!

Reply:
My thinking was just that we would for sure introduce memory resource types that would not be host-accessible. We would just never use those with `vecmem::vector`, only with some other, simpler constructs. (Like [Acts::Cuda::device_array](https://github.com/acts-project/acts/blob/master/Plugins/Cuda/include/Acts/Plugins/Cuda/Utilities/Arrays.hpp).)

So while the ability to pair the wrong resource type with a given higher-level type will be ever-present, I just don't know how we could avoid this. At least in `vecmem::vector`, where it really counts...